docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -72,6 +72,13 @@ space. By default {es} will automatically create a mapping for every field in
 every document it indexes, but you can switch off this behaviour to
 <<explicit-mapping,take control of your mappings>>.
 
+Moreover every segment requires a small amount of heap memory for each mapped
+field. This per-segment-per-field heap overhead includes a copy of the field
+name, encoded using UTF-8. Usually this is not noticeable, but you may need to
+account for this overhead if you have very long field names or high field
+counts in any of your mappings, or if any of your shards have high segment
```

Comment:
I think it is the combination of many segments with many fields (possibly long names) that makes this a problem. Can we mention that somehow, perhaps rephrasing to:

`... account for this overhead if you have very long field names or high field counts in combination with high segment counts
```

Reply:
In principle it could really be any combination of the above but yes it's only really not covered by the per-field overhead at high segment counts (or crazy field name lengths). Fixed in 913e68b1748.